CodeStarterCamp_Week2/LottoDatabase.swift

```diff
@@ -0,0 +1,34 @@
+//
+//  LottoDatabase.swift
+//  CodeStarterCamp_Week2
+//
+//  Created by QL Jeong on 2023/08/25.
+//
+
+import Foundation
+
+@discardableResult func addWinNumbers(_ winNumbers: [Int]) -> Int {
+    let currentRound = winNumbersList.count + 1
+    let key = makeKey(from: currentRound)
+    winNumbersList.updateValue(winNumbers, forKey: key)
+
+    return currentRound
+}
+
+func getWinNumbers(at round: Int) -> [Int]? {
+    let key = makeKey(from: round)
+
+    return winNumbersList[key]
+}
+
+func getAllWinNumbers() -> [String:[Int]] {
+    return winNumbersList
+}
+
+// MARK: - Private
+
+fileprivate var winNumbersList = [String:[Int]]()
```

Comment:
딕셔너리의 타입을 나타낼 때 보통 : 뒤에 공백을 한 칸 넣어줍니다.
```suggestion
fileprivate var winNumbersList = [String: [Int]]()
```
혹시 QL님은 type inference 와 type annotation을 사용하는 기준을 갖고 계실까요?!
저는 개인적으로  collection 타입 또는 타입의 프로퍼티 등에는 type annotation을 사용하는 편입니다. 🤔
(함수 내부 등 지역적으로 사용되는 상수/변수의 경우는 type inference를 사용합니다 👀)

```swift
fileprivate var winNumbersList: [String: Int] = [:]
```
또는 .init을 사용하기도 합니다.
```swift
fileprivate var winNumbersList: [String: Int] = .init()
```
개인적으로 보기 좋아서 사용하는 방법이니 참고만 해주세요~!

Reply:
type 에대한 inference 와 annotation 은 여러 사람들을 신경을 긁어대는 문제이긴 하지요.
`너가 잘못인건 아닌데 내 스타일과는 달러` 라면서요.

저는 초기화 또는 값을 대입하는 작업이
선언과 동시에 일어나지 않은 경우에는 annotation 을 사용 합니다.
뭐 에초에 이건 inference 으로는 방법이 없지요.
```swift
enum City {
    case seoul, daejeon, daegu, busan, jeju
}

var name : String
var addressCity: City

name = "QL"
addressCity = .seoul
```

또다른 한가지는 Float/Double 에서 자주 일어나는 문제긴 한데
String/Character 에서도 일어나긴 합니다.
어떠한 값이 다수의 Type 에 동시 포함될수 있다면 명시하도록 합니다.
```swift
var gender : Character = "M" // 확실하게 String 이 아니도록 명시
var height : Double = 0 // 확실하게 Double 로 명시
height = 172.0
```

즉 그렇다면 저는 inference 을 선호 하고
특수한 경우에만 annotation 으로 명시하게 되는 것이라고 생각 합니다.

말씀 해주신 두가지 초기화 방법은 모두다 알고 있는 방식입니다.
collection 의 경우 저는 글자수 (그거 몇자 안되긴 하지만) 가
줄어들어서 inference 을 사용 하고 있습니다.
```swift
var winNumbersList = [String: [Int]]()
var winNumbersList: [String: Int] = [:]
var winNumbersList: [String: Int] = .init()
```
또한 다른 선언들과 똑같이 늘어 놓았을때 = 기준으로 좌우를 판단 하기 편해서 이기도 합니다.
```swift
var winNumbersList = [String: [Int]]()
var lastRoundNumber = 0
var isCarriedOver = false
```

앞으로는 띄어쓰기에 주의 하겠습니다.
감사합니다.